src/main/java/com/eppo/sdk/EppoClient.java

```diff
@@ -187,22 +187,50 @@ private Optional<EppoValue> determineAndLogBanditAction(VariationAssignmentResul
                 actionAttributes = assignmentOptions.get(actionString);
             }
 
-            // TODO FF-1546: break out logging the attributes by numeric and categorical
+            Map<String, Double> subjectNumericAttributes = numericAttributes(assignmentResult.getSubjectAttributes());
+            Map<String, String> subjectCategoricalAttributes = categoricalAttributes(assignmentResult.getSubjectAttributes());
+            Map<String, Double> actionNumericAttributes = numericAttributes(actionAttributes);
+            Map<String, String> actionCategoricalAttributes = categoricalAttributes(actionAttributes);
+
             this.eppoClientConfig.getBanditLogger().logBanditAction(new BanditLogData(
               assignmentResult.getExperimentKey(),
               banditName,
               assignmentResult.getSubjectKey(),
-              assignmentResult.getSubjectAttributes(),
               actionString,
-              actionAttributes,
               actionProbability,
-              modelVersionToLog
+              modelVersionToLog,
+              subjectNumericAttributes,
+              subjectCategoricalAttributes,
+              actionNumericAttributes,
+              actionCategoricalAttributes
             ));
         }
 
         return Optional.of(actionValue);
     }
 
+    private Map<String, Double> numericAttributes(EppoAttributes eppoAttributes) {
+        if (eppoAttributes == null) {
+            return Map.of();
+        }
+        return eppoAttributes.entrySet().stream().filter(e -> e.getValue().isNumeric()
+        ).collect(Collectors.toMap(
+          Map.Entry::getKey,
+          e -> e.getValue().doubleValue())
+        );
+    }
+
+    private Map<String, String> categoricalAttributes(EppoAttributes eppoAttributes) {
+        if (eppoAttributes == null) {
+            return Map.of();
+        }
+        return eppoAttributes.entrySet().stream().filter(e -> !e.getValue().isNumeric() && !e.getValue().isNull()
+        ).collect(Collectors.toMap(
+          Map.Entry::getKey,
+          e -> e.getValue().toString())
```

Comment:
`toString()` is used over `stringValue()` because we want the string representation of non-string-typed EppoValues (e.g., Booleans)